### PR TITLE
Dp crud

### PR DESCRIPTION
**Pull request by Dpalazzari**
@drod1000 @njgheorghita @Laszlo-JFLMTCO

### CODE REVIEW

File: app/models/outing.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Outing < ApplicationRecord
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :image_url, presence: true
+  validates :base_cost, presence: true
+
```
Comment (njgheorghita) [inline app/models/outing.rb:3]: refactor down to one line?

File: app/models/politician.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Politician < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+  validates :party, presence: true
+  validates :image, presence: true, uniqueness: true
+  validates :multiplier, presence: true
+
```
Comment (njgheorghita) [inline app/models/politician.rb:4]: ditto?

**Comment by njgheorghita** [comment]
looks good to me, merge on!

**Comment by Dpalazzari** [comment]
🚀

File: spec/models/outing_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      end
+
+      it "is invalid without an image" do
+        outing = Outing.create(title: "thing", description: "thingy3", base_cost: 4)
+
+        expect(outing).to be_invalid
+      end
+
+      it "is invalid without base cost" do
+        outing = Outing.create(title: "thing", image_url: "thingy1", image_url: "bob")
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply spec/models/outing_spec.rb]: @drod1000 Let's fix the duplicated hash key issue before merging into Master. @njgheorghita @drod1000

**Comment by drod1000** [comment]
Looks Good!

File: spec/models/politician_spec.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+require "rails_helper"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply spec/models/politician_spec.rb:1]: @drod1000 confirmed this message is not applicable to our implementation due to Ruby version we use vs what Ruby version Hound is using. FYI, @njgheorghita @Dpalazzari

File: spec/models/politician_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  describe 'validations' do
+    context 'invalid attributes' do
+      it 'is invalid without a name' do
+        politician = Politician.create(party: 'de', multiplier: 2, image: 'th')
+
+        expect(politician).to be_invalid
+      end
+
+      it 'has a unique name' do
+        politician = Politician.create(name: 'Dan', party: 'd', multiplier: 2, image: 't')
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply spec/models/politician_spec.rb:13]: This error message is incorrect. This test case is necessary to test uniqueness evaluation of the name column when creating new row in our Politicians table. FYI @Dpalazzari @njgheorghita @drod1000

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
+    t.string   "party"
+    t.integer  "multiplier"
+    t.string   "image"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "outings", "politicians"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:38]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
+    t.string   "party"
+    t.integer  "multiplier"
+    t.string   "image"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:35]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    t.datetime "updated_at",    null: false
+    t.integer  "politician_id"
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
+    t.string   "party"
+    t.integer  "multiplier"
+    t.string   "image"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:33]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    t.datetime "created_at",    null: false
+    t.datetime "updated_at",    null: false
+    t.integer  "politician_id"
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
+    t.string   "party"
+    t.integer  "multiplier"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:32]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    t.string   "image_url"
+    t.datetime "created_at",    null: false
+    t.datetime "updated_at",    null: false
+    t.integer  "politician_id"
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
+    t.string   "party"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:31]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    t.integer  "base_cost"
+    t.string   "image_url"
+    t.datetime "created_at",    null: false
+    t.datetime "updated_at",    null: false
+    t.integer  "politician_id"
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
+    t.string   "name"
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:30]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    t.string   "description"
+    t.integer  "base_cost"
+    t.string   "image_url"
+    t.datetime "created_at",    null: false
+    t.datetime "updated_at",    null: false
+    t.integer  "politician_id"
+    t.index ["politician_id"], name: "index_outings_on_politician_id", using: :btree
+  end
+
+  create_table "politicians", force: :cascade do |t|
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:29]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

File: db/schema.rb [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    t.string   "name"
+    t.string   "party"
+    t.integer  "multiplier"
+    t.string   "image"
+    t.datetime "created_at", null: false
```
[the remark this answers is not in the document]
Comment (Laszlo-JFLMTCO) [reply db/schema.rb:34]: Rails is generating `shema.rb` file. Based on our joint review we should ignore this warning. FYI @drod1000 @njgheorghita @Dpalazzari

**Pull request merged**